modules/analysis/cohort.py

```
@@ -106,7 +122,145 @@ def get_user_info_by_year(info_db_no, origin_table, year):
+
+    p_retention = calculate_monthly_retention_by_churn(group_total_counts['P'], {m: monthly_dropout_counts[m]['P'] for m in range(1, 13)})
+    c_retention = calculate_monthly_retention_by_churn(group_total_counts['C'], {m: monthly_dropout_counts[m]['C'] for m in range(1, 13)})
+    l_retention = calculate_monthly_retention_by_churn(group_total_counts['L'], {m: monthly_dropout_counts[m]['L'] for m in range(1, 13)})
+
+    try:
+        save_pcl_csv_to_s3(info_db_no, p_retention, c_retention, l_retention)
+        return jsonify({
+            "success": True,
+            "message": "CSV 파일이 S3에 저장되었습니다.",
+        }), 200
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)